packages/astro/src/core/render/dev/css.ts

```diff
@@ -64,14 +67,24 @@ export async function getStylesForURL(
 			}
 			const ext = path.extname(importedModule.url).toLowerCase();
 			if (STYLE_EXTENSIONS.has(ext)) {
-				// NOTE: We use the `url` property here. `id` would break Windows.
-				importedCssUrls.add(importedModule.url);
+				if (
+					mode === 'development' // only inline in development
+					&& typeof importedModule.ssrModule?.default === 'string' // ignore JS module styles
```

Comment:
A few use cases still build styles to JS modules rather than plain CSS strings, the main one I came across being CSS modules